### DISCUSSION TITLE
Support installing entire napari-hub collections in one go

### DISCUSSION THREAD

**Discussion by seankmartin** [chanzuckerberg/napari-hub#1066] [Ideas]
## Description
Currently, collections are serving only as a curated list, which users can use to install plugins from one by one.
It might be nice to allow users to install the whole collection in one go. For instance, this might allow users to recommend a full set of plugins to use alongside their plugin, replicating similar behaviour to the following:

To install the recommended companion plugins for our plugin run:

```
python -m pip install myplugin[recommended]
```

A potential implementation of this is an "export collection as a pip requirements file" style of thing.

### Potential difficulty in adopting the above style to a collection
If I wanted to recommend some plugins to use alongside my plugin, there is a possibility I might want to pin specific versions of some of those plugins. Specifying version constraints does not seem to be possible in collections at the moment.

**Comment by Nadalyn-CZI** [comment]
I think this is a good idea. Could people be allowed to select only the plugins they want? i.e. A collection has 25 plugins.  Maybe the user is interested in only 10 of those.  Could they be allowed to select only the ones they want out of that collection? 

This begs the question, could a person select certain plugins whether the plugins were in the same collection or not?

**Comment by dgmccart** [comment]
The Plugin Collections feature was deprecated in August 2023, so this idea is no longer applicable.
